src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -625,6 +632,36 @@ private async Task SendStreamItems<T>(string streamId, ChannelReader<T> reader,
             await SendWithLock(CompletionMessage.WithError(streamId, responseError));
         }
 
+#if NETCOREAPP3_0
+        // this is called via reflection using the `_sendStreamItems` field
+        private async Task SendIAsyncEnumerableStreamItems<T>(string streamId, IAsyncEnumerable<T> stream, CancellationToken token)
+        {
+            Log.StartingStream(_logger, streamId);
+
+            var combinedCts = CancellationTokenSource.CreateLinkedTokenSource(_uploadStreamToken, token);
+
+            stream = AsyncEnumerableAdapters.MakeCancelableTypedAsyncEnumerable(stream, combinedCts);
```

Comment:
Consider making a common stream method so we don't duplicate the streaming code, for example
```c#
async Task CommonStreaming(Func<CancellationTokenSource, Task> createAndConsumeStream)
{
    Log.StartingStream(_logger, streamId);
    var combinedCts = CancellationTokenSource.CreateLinkedTokenSource(_uploadStreamToken, token);

    try
    {
        await createAndConsumeStream(combinedCts);
    }
    catch (OperationCanceledException)
    {
        Log(...)
        responseError...
    }

    Log(...);
    await SendWithLock(...);
}
```